### PR TITLE
core/thread.c: change thread_getstatus return type

### PR DESCRIPTION
**Pull request by JulianHolzwarth**
### Contribution description

`thread_getstatus()` return type is after the change `thread_status_t` instead of an `int`. To compare it to a status easier.

### CODE REVIEW

**Comment by JulianHolzwarth** [comment]
@MichelRottleuthner

**Comment by benpicco** [comment]
Murdock is not happy - might be it found an actual bug.

**Comment by JulianHolzwarth** [comment]
> Murdock is not happy - might be it found an actual bug.

thanks

File: sys/cpp11-compat/thread.cpp [hunk 1]
```diff
@@ -42,7 +42,7 @@ void thread::join() {
   }
   if (joinable()) {
     auto status = thread_getstatus(m_handle);
-    if (status != (int)STATUS_NOT_FOUND && status != STATUS_STOPPED) {
+    if (status != STATUS_NOT_FOUND && status != STATUS_STOPPED) {
```
Comment (benpicco) [inline sys/cpp11-compat/thread.cpp]: better add some parenthesis here to make the evaluation order clear.

Comment (JulianHolzwarth) [reply sys/cpp11-compat/thread.cpp]: What do you mean? The order should not matter. 
`status != -1 && status != 0`

Comment (benpicco) [reply sys/cpp11-compat/thread.cpp]: Oh, nvm I thought it was `&` instead of `&&`.

**Pull request merged**